spacy/cli/debug_data.py

```diff
@@ -670,6 +671,40 @@ def debug_data(
                 f"Found {gold_train_data['n_cycles']} projectivized train sentence(s) with cycles"
             )
 
+    if "trainable_lemmatizer" in factory_names:
+        msg.divider("Trainable Lemmatizer")
+        trees_train: Set[str] = gold_train_data["lemmatizer_trees"]
+        trees_dev: Set[str] = gold_dev_data["lemmatizer_trees"]
+
+        # This is necessary context when someone is attempting to interpret whether the
+        # number of trees exclusively in the dev set is meaningful.
+        msg.info(f"{len(trees_train)} lemmatizer trees generated from training data.")
+        msg.info(f"{len(trees_dev)} lemmatizer trees generated from dev data.")
+
+        dev_not_train = trees_dev - trees_train
+        if len(dev_not_train) != 0:
+            msg.warn(
+                f"{len(dev_not_train)} lemmatizer trees were found exclusively in the dev data."
+            )
+        else:
+            # Would we ever expect this case? It seems like it would be pretty rare,
+            # and we might actually want a warning?
+            msg.good("All trees in dev data present in training data.")
+
+        if gold_train_data["n_low_cardinality_lemmas"] > 0:
+            msg.warn(
+                f"{gold_train_data['n_low_cardinality_lemmas']} docs with 1 or 0 unique lemmas."
```

Comment:
This reports if a doc is not annotated at all. But I guess another possibility is that only a subset of tokens are annotated, which wouldn't be reported now, as long as there are enough (unique) annotations in the document. Maybe it's worthwhile to report the number of tokens without a lemma annotation?

Reply:
Usually our goal is to support training from partial annotation, so this isn't a check that we do for other components or that we necessarily want to do here. I think this intended to be parallel to "we found 47 tags in the training data" as a check for training data that's not going to work for this component at all. (Does the edit tree lemmatizer skip unlabeled examples?)

Reply:
Ah, right.

> Does the edit tree lemmatizer skip unlabeled examples?

It does:

https://github.com/explosion/spaCy/blob/6b78135b9e158e5bc02e39c1a73ef28bb360a44f/spacy/pipeline/edit_tree_lemmatizer.py#L131

Reply:
Okay, then I think checking for absolutely no annotation (usually a data conversion issue) is what we want here.

Reply:
Just to clarify: Do we want to change this to check how many documents have one or more `None` values as the `lemma` attribute? Or to show how many lemmas across all documents are without an annotation?

Reply:
I've updated it to check if all lemmas are `0` for a doc and if they are, increase a counter by 1 (and do not loop through the tokens and collect the string tree representations). 

@danieldk A question on this: if a doc doesn't have an annotation is the value `None` or `0`? In the lemmatizer code linked above, it checks whether the attribute is `None`, but if I create a doc with without lemmas, they're all `0`. Is it something different with annotations that they have a value of `None` when they don't exist? Either way, I'm not clear what we need to check on here (`None` or `0`).

```python
import spacy

nlp = spacy.blank("en")
doc = nlp("This is a document to serialize.")
print([t.lemma for t in doc])
# [0, 0, 0, 0, 0, 0, 0]
```

Reply:
> @danieldk A question on this: if a doc doesn't have an annotation is the value `None` or `0`? In the lemmatizer code linked above, it checks whether the attribute is `None`, but if I create a doc with without lemmas, they're all `0`. Is it something different with annotations that they have a value of `None` when they don't exist? Either way, I'm not clear what we need to check on here (`None` or `0`).

We should use `Token.lemma` for checking absence and when a lemma is absent, the value is `0`. It cannot be `None`, since `Token.lemma` just passes through the C representation, which is integral:

https://github.com/explosion/spaCy/blob/8e558095a1223fcb81e0abeda46e1e6975729254/spacy/structs.pxd#L44

(`attr_t` is an `uint64_t`).

I think that the fragment that I linked for the edit tree lemmatizer is incomplete, since `get_aligned` returns `None` for failed alignments, but empty strings for absent lemma annotations (like `Token.lemma_` would). I'll open a PR for that.

Reply:
Resolved in latest commit, which distinguishes the complete and partial annotation cases and now has a separate counter for both. The message is a warn if there are a non-zero amount of docs with absolutely no annotations, and an info for the number of docs with partial annotations (and a good if both are 0).